lib/provider/persistence_provider.dart

```diff
@@ -64,6 +70,11 @@ class PersistenceService {
     return PersistenceService._(prefs);
   }
 
+  static Future<PersistenceService> getPersistence() async {
```

Comment:
`riverpod` is used as the dependency injection tool so you can get the instance by calling

`final persistenceService = ref.watch(persistenceProvider);`

The widget must be a `ConsumerWidget` / `ConsumerStatefulWidget` though. So either you can change the `WindowWatcher` from a `StatefulWidget` to a `ConsumerStatefulWidget` or you can get the `persistenceService` inside `main.dart`.

Reply:
Noted! I'll rewrite a couple of things and make a new commit soon.